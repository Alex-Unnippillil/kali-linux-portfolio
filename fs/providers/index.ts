export * from './types';
export { MemoryFileSystemProvider } from './memory';
export { ZipFileSystemProvider } from './zip';
