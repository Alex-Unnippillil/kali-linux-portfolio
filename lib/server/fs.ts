import 'server-only';
import fs from 'fs';

export const promises = fs.promises;

export default fs;
