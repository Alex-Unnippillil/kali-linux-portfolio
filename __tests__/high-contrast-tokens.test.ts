import fs from 'fs';
import path from 'path';

const readTokens = () =>
  fs.readFileSync(path.join(process.cwd(), 'styles', 'tokens.css'), 'utf8');

const extractBlock = (css: string, selector: string) => {
  const pattern = new RegExp(`${selector}\\s*{([^}]*)}`, 'm');
  const match = css.match(pattern);
  if (!match) {
    throw new Error(`Could not find block for ${selector}`);
  }
  return match[1].trim();
};

const extractMediaRoot = (css: string) => {
  const pattern = /@media\s*\(prefers-contrast:\s*more\)\s*{[\s\S]*?:root\s*{([\s\S]*?)}\s*}/m;
  const match = css.match(pattern);
  if (!match) {
    throw new Error('Could not find prefers-contrast media block');
  }
  return match[1].trim();
};

describe('high contrast tokens', () => {
  const css = readTokens();
  const classBlock = extractBlock(css, '\\.high-contrast');
  const mediaBlock = extractMediaRoot(css);

  test('class block matches snapshot', () => {
    expect(classBlock).toMatchSnapshot();
  });

  test('media query block matches snapshot', () => {
    expect(mediaBlock).toMatchSnapshot();
  });

  test('media query tokens mirror class tokens', () => {
    const normalize = (block: string) =>
      block
        .split('\n')
        .map((line) => line.trim())
        .filter(Boolean)
        .sort();
    expect(normalize(mediaBlock)).toEqual(normalize(classBlock));
  });
});
