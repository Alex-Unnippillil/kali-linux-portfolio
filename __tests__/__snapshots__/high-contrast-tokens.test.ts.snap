// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`high contrast tokens class block matches snapshot 1`] = `
"--color-bg: #000000;
  --color-text: #ffffff;
  --color-ub-grey: #000000;
  --color-ub-cool-grey: #000000;
  --color-ubt-grey: #ffffff;
  --color-ubt-cool-grey: #ffffff;
  --color-ub-orange: #ffff00;
  --color-ub-lite-abrgn: #00ffff;
  --color-ub-border-orange: #ffff00;
  --color-primary: #ffff00;
  --color-secondary: #111111;
  --color-accent: #ffff00;
  --color-muted: #111111;
  --color-surface: #000000;
  --color-border: #ffffff;
  --color-focus-ring: #ffff00;
  --color-selection: #00ffff;
  --color-control-accent: #ffff00;
  --color-window-border: #ffffff;
  --color-window-accent: #ffff00;
  --color-window-title: #000000;
  --kali-bg: #000000;
  --kali-bg-solid: #000000;
  --kali-panel: #000000;
  --kali-panel-border: #ffffff;
  --kali-panel-highlight: rgba(255, 255, 255, 0.15);
  --game-color-secondary: #ffff00;
  --game-color-success: #00ffff;
  --game-color-warning: #ff00ff;
  --game-color-danger: #ff0000;"
`;

exports[`high contrast tokens media query block matches snapshot 1`] = `
"--color-bg: #000000;
    --color-text: #ffffff;
    --color-ub-grey: #000000;
    --color-ub-cool-grey: #000000;
    --color-ubt-grey: #ffffff;
    --color-ubt-cool-grey: #ffffff;
    --color-ub-orange: #ffff00;
    --color-ub-lite-abrgn: #00ffff;
    --color-ub-border-orange: #ffff00;
    --color-primary: #ffff00;
    --color-secondary: #111111;
    --color-accent: #ffff00;
    --color-muted: #111111;
    --color-surface: #000000;
    --color-border: #ffffff;
    --color-focus-ring: #ffff00;
    --color-selection: #00ffff;
    --color-control-accent: #ffff00;
    --color-window-border: #ffffff;
    --color-window-accent: #ffff00;
    --color-window-title: #000000;
    --kali-bg: #000000;
    --kali-bg-solid: #000000;
    --kali-panel: #000000;
    --kali-panel-border: #ffffff;
    --kali-panel-highlight: rgba(255, 255, 255, 0.15);
    --game-color-secondary: #ffff00;
    --game-color-success: #00ffff;
    --game-color-warning: #ff00ff;
    --game-color-danger: #ff0000;"
`;
